interfaces/repo_test.go

```diff
@@ -1334,15 +1334,24 @@ func (s *DisconnectSnapSuite) TestNotConnected(c *C) {
 	c.Check(affected, HasLen, 0)
 }
 
+func snapNamesFor(snapInfoSet map[*snap.Info]bool) []string {
+	snapNames := make([]string, len(snapInfoSet))
```

Comment:
this is wrong: either make it with length 0 and `len(snapInfoSet)` capacity (and use `append` as you do below), or make it with this length and use direct assignment below. This way you're ending up with a list 2× the length, with the first half full of empty strings.